napari/layers/surface/surface.py

```diff
@@ -421,37 +498,69 @@ def _get_state(self):
                 'data': self.data,
                 'wireframe': self.wireframe.dict(),
                 'normals': self.normals.dict(),
+                'texture': self.texture,
+                'texcoords': self.texcoords,
+                'vertex_colors': self.vertex_colors,
             }
         )
         return state
 
+    def _slice_associated_data(
+        self,
+        data: np.ndarray,
+        vertex_ndim: int,
+        dims: int = 1,
+    ) -> Union[List[Any], np.ndarray]:
+        """Return associated layer data (e.g. vertex values, colors) within
+        the current slice.
+        """
```

Comment:
```suggestion
    @staticmethod
    def _slice_associated_data(
        data: np.ndarray,
        vertex_ndim: int,
        dims: int = 1,
    ) -> Union[List[Any], np.ndarray]:
        """Return associated layer data (e.g. vertex values, colors) within
        the current slice.
        """
```

Reply:
I don't think this can simply be static because it accesses `self._slice_indices[:-vertex_ndim]`.

Reply:
Lol, I even searched for `self` in the text and somehow missed that xD